### DISCUSSION TITLE
Duplicate IP address found in global table

### DISCUSSION THREAD

**Discussion by jb606** [netbox-community/ansible_modules#1190] [Q&A] [answered]
I've been trying to figure this out for a while and hope someone can point me in the right direction.  I've written a playbook that will "interview" a machine and put its information in to Netbox and I'm hung up when it goes to add and assign the IP address to an interface.  Most of our IP addresses are already in Netbox (not assigned to devices) so when the netbox.netbox_ip_address module runs it errors out with the message "Duplicate IP address found in global table: x.x.x.x/xx".  Makes sense, the address was in there but I want Ansible to ignore that and just add x.x.x.x/xx to an interface.  Any ideas?  

`- name: "Adding IP Address"`
`      netbox.netbox.netbox_ip_address:`
`        netbox_url: "{{ netbox_url }}"`
`        netbox_token: "{{ netbox_token }}"`
`        validate_certs: False`
`        state: present`
`        data:`
`          address: "{{hostvars[ip].ansible_default_ipv4.address}}/{{hostvars[ip].ansible_default_ipv4.prefix}}"`
`          dns_name: "{{hostvars[ip].ansible_hostname}}"`
`          assigned_object:`
`                device: "{{hostvars[ip].ansible_hostname}}"`
`                name: "{{ hostvars[ip].ansible_default_ipv4.alias }}"`
`

**Comment by sc68cal** [comment, marked as answer]
I don't believe the logic around `assigned_object` is complex enough to handle the update of an existing IP address, to associate it with a device. I am a little fuzzy but I seem to recall encountering this and having to delete IP addresses, then re-create them using `netbox_ip_address` module with the `assigned_object` correctly set

**Reply by abcd678** [reply]
Is this a netbox limitation or a limitation of the ansible module? Unless I'm missing some other way of handling it this would make it unfeasible to automate this process while retaining readable changelogs within netbox.
